kubectl-cloudflow/cmd/deploy.go

```diff
@@ -81,32 +105,195 @@ the stored credentials.
+			if cmd.Flag("image").Changed == false && cmd.Flag("blueprint").Changed == false {
+				util.LogAndExit("Please specify either the blueprint file or the image path")
+			}
+	    if (cmd.Flag("image").Changed == true) && (cmd.Flag("blueprint").Changed == true) {
+		    util.LogAndExit("%s", "Cannot specify both blueprint and image in command")
+	    }
+			if cmd.Flag("image").Changed == true {
+				fmt.Printf("Deploying image %s\n", cmd.Flag("image").Value.String())
+			} else if cmd.Flag("blueprint").Changed == true {
+				fmt.Printf("Deploying blueprint %s\n", cmd.Flag("blueprint").Value.String())
+			}
+		},
+		Run: deployOpts.deployImpl,
 	}
+	deployOpts.cmd.Flags().StringVarP(&deployOpts.blueprintFilePath, "blueprint", "b", "", "blueprint file path.")
+	deployOpts.cmd.Flags().StringVarP(&deployOpts.imagePath, "image", "i", "", "docker image path.")
 	deployOpts.cmd.Flags().StringVarP(&deployOpts.username, "username", "u", "", "docker registry username.")
 	deployOpts.cmd.Flags().StringVarP(&deployOpts.password, "password", "p", "", "docker registry password.")
 	deployOpts.cmd.Flags().BoolVarP(&deployOpts.passwordStdin, "password-stdin", "", false, "Take the password from stdin")
 	deployOpts.cmd.Flags().StringArrayVar(&deployOpts.volumeMounts, "volume-mount", []string{}, "Accepts a key/value pair separated by an equal sign. The key should be the name of the volume mount, specified as '[streamlet-name].[volume-mount-name]'. The value should be the name of an existing persistent volume claim.")
+	deployOpts.cmd.Flags().StringToIntVar(&deployOpts.replicas, "scale", map[string]int{}, "Accepts key/value pairs for replicas per streamlet")
 
 	rootCmd.AddCommand(deployOpts.cmd)
 }
 
 func (opts *deployOptions) deployImpl(cmd *cobra.Command, args []string) {
 	version.FailOnProtocolVersionMismatch()
 
-	imageRef := args[0]
+	if cmd.Flag("image").Changed == true {
+		deployImage(cmd, opts, args)
+	} else {
+		deployBlueprint(cmd, opts, args)
+	}
+}
+
+func deployBlueprint(cmd *cobra.Command, opts *deployOptions, args []string) {
+	// read blueprint file and load its content
+	blueprintFile := cmd.Flag("blueprint").Value.String()
+	content, err := ioutil.ReadFile(blueprintFile)
+	if err != nil {
+		util.LogAndExit("Cannot read blueprint file: Error : %s", err.Error())
+	}
+	// this needs to be replaced with calls from blueprint package once verify blueprint is in
+	blueprintConfig, err := loadBlueprintConfigContent(string(content[:]))
+	if err != nil {
+		util.LogAndExit("Cannot load blueprint file information: Error : %s", err.Error())
+	}
+
+	// get the images from the blueprint
+	imagesByName := blueprintConfig.GetNode("blueprint.images").GetObject().Items()
+
+	var imageRefs []deploy.ImageReference
+	var dockerRegistryURL string
+
+	for _, imageRef := range imagesByName {
+		ref, err := parseImageReference(imageRef.GetString())
+		if err != nil {
+			util.LogAndExit("Cannot parse image reference from blueprint: Error : %s", err.Error())
```

Comment:
Minor, you could collect all errors and tell the user at once about all image references that could not be parsed, instead of finding them one by one.

Reply:
done ..